test/unit/auth_sources/auth_source_ldap_test.rb

```diff
@@ -167,7 +167,16 @@ def setup
     end
 
     test 'update_usergroups calls refresh_ldap if entry belongs to some group' do
-      ExternalUsergroup.expects(:find_by_name).with('ipausers').returns(ExternalUsergroup.new)
+      @auth_source_ldap.expects(:external_usergroups).returns(ExternalUsergroup)
+      ExternalUsergroup.expects(:where).with('lower(name) = ?', 'ipausers').returns([ExternalUsergroup.new])
+      ExternalUsergroup.any_instance.expects(:refresh)
+      @auth_source_ldap.send(:update_usergroups, 'test')
+    end
+
+    test 'update_usergroups matches LDAP gids with external user groups case insensitively' do
```

Comment:
I think this test is missing a good assertion to check that the match really worked.  The only imprecise assertion is the expects(:refresh) right now.